src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardSamlBundle/Security/Authentication/Provider/SamlProvider.php

```diff
@@ -93,7 +93,12 @@ public function authenticate(TokenInterface $token)
 
         $nameId = $translatedAssertion->getNameID();
         $email = $this->getSingleStringValue('mail', $translatedAssertion);
-        $commonName = $this->getSingleStringValue('commonName', $translatedAssertion);
+
+        try {
+            $commonName = $this->getSingleStringValue('commonName', $translatedAssertion);
+        } catch (BadCredentialsException $e) {
```

Comment:
The exception thrown does not really reflect the issue at hand. But that should have been noted when this feature was implemented 2 years ago. If you feel like it, maybe add a new exception named something like UnknownAttributeRequestedException.

Reply:
Good suggestion, created a new event for this.